src/Type/TypehintHelper.php

```diff
@@ -78,9 +80,6 @@ public static function decideType(
 		?Type $phpDocType,
 	): Type
 	{
-		if ($phpDocType !== null && $type->isNull()->no()) {
-			$phpDocType = TypeCombinator::removeNull($phpDocType);
-		}
```

Comment:
I think the changes in TypehintHelper can stay as they were.

Reply:
sure, I updated to keep TypeHintHelper as they were 👍 https://github.com/phpstan/phpstan-src/pull/3979/commits/4e820816ad78608d7f2ef8bbb9f322a878d45fa5